spec/controllers/state_file/questions/nj_homeowner_eligibility_controller_spec.rb

```diff
@@ -0,0 +1,182 @@
+require "rails_helper"
+
+RSpec.describe StateFile::Questions::NjHomeownerEligibilityController do
+  let(:intake) { create :state_file_nj_intake }
+  before do
+    sign_in intake
+  end
+
+  describe "#edit" do
+    render_views
+    it 'succeeds' do
+      get :edit
+      expect(response).to be_successful
+    end
+
+    context "when a user is a homeowner" do
+      let(:form_params) {
+        {
+          state_file_nj_homeowner_eligibility_form: {
+            homeowner_home_subject_to_property_taxes: :yes,
+            homeowner_more_than_one_main_home_in_nj: :no,
+            homeowner_shared_ownership_not_spouse: :yes,
+          }
+        }
+      }
+
+      it "saves the checkbox selections" do
+        post :update, params: form_params
+
+        intake.reload
+        expect(intake.homeowner_home_subject_to_property_taxes).to eq "yes"
+        expect(intake.homeowner_more_than_one_main_home_in_nj).to eq "no"
+        expect(intake.homeowner_shared_ownership_not_spouse).to eq "yes"
+      end
+    end
+  end
+
+  describe "#show?" do
+    context "when indicated that they own" do
+      let(:intake) { create :state_file_nj_intake, household_rent_own: "own" }
+      it "shows" do
+        expect(described_class.show?(intake)).to eq true
+      end
+    end
+
+    context "when indicated that they rent" do
+      let(:intake) { create :state_file_nj_intake, household_rent_own: "rent" }
+      it "does not show" do
+        expect(described_class.show?(intake)).to eq false
+      end
+    end
+
+    context "when indicated neither rent nor own" do
+      let(:intake) { create :state_file_nj_intake, household_rent_own: "neither" }
+      it "does not show" do
+        expect(described_class.show?(intake)).to eq false
+      end
+    end
+
+    context "when indicated both rent and own" do
+      let(:intake) { create :state_file_nj_intake, household_rent_own: "both" }
+      it "does not show" do
+        expect(described_class.show?(intake)).to eq false
+      end
+    end
+  end
+
+  describe "#next_path" do
+    describe "hard no ineligible states" do
+      context "when NO to homeowner_home_subject_to_property_taxes" do
```

Comment:
[dust] Maybe instead of using the field name, it makes sense to make the test context human readable for additional context, since the field name is in the test anyway

Reply:
Good point. I think I'm just finding it hard to remember what some fields mean without referring to en.yml-- not for any fault in naming, but bc this is such dense subject matter! 😅 If you wanted to change it, maybe something like "when a taxpayer paid neither rent nor property taxes"?